🐛 Import path mismatch: Value objects reference specific error files but YAML creates barrel export

## 🐛 Bug Report

**Discovered During**: Dogfooding Experiment 003 - Phase 5 (Evaluation)
**Command**: `/05-evaluate-layer-results`
**File**: `spec/001-product-catalog-management/domain/shared-implementation.yaml`
**Severity**: CRITICAL (causes runtime errors)
**Expected RLHF Score**: -1 (RUNTIME ERROR)

---

## 📋 Problem Description

The generated YAML plan has an import path mismatch between value object imports and error file structure:

- **Value Objects Import**: `from '../errors/invalid-sku-error'` (expects specific file)
- **Actual File Created**: `errors/index.ts` (barrel export containing all errors)
- **Result**: TypeScript compilation error "Cannot find module '../errors/invalid-sku-error'"

---

## 🔍 Affected Steps

| Step | Component | Import Statement | Expected File | Actual File |
|------|-----------|------------------|---------------|-------------|
| Step 4 | SKU Value Object | `from '../errors/invalid-sku-error'` | `errors/invalid-sku-error.ts` | `errors/index.ts` |
| Step 5 | Price Value Object | `from '../errors/invalid-price-error'` | `errors/invalid-price-error.ts` | `errors/index.ts` |
| Step 6 | InventoryLevel Value Object | `from '../errors/invalid-inventory-error'` | `errors/invalid-inventory-error.ts` | `errors/index.ts` |

All three value objects will fail with "Module not found" errors.

---

## 🧪 Evidence from Evaluation

```
Step 4 (SKU):
import { InvalidSKUError } from '../errors/invalid-sku-error'

Step 8 (Errors file path):
path: 'product-catalog/src/features/product-catalog-management/shared/domain/errors/index.ts'

Step 8 (Errors template):
export class InvalidSKUError extends Error { ... }
export class InvalidPriceError extends Error { ... }
export class InvalidInventoryError extends Error { ... }
export class ProductNotFoundError extends Error { ... }
```

TypeScript will look for `errors/invalid-sku-error.ts` but will find `errors/index.ts` instead.

---

## 🔧 Proposed Solutions

### Option 1: Fix Imports (Recommended) ✅

Change all error imports in value objects to use barrel exports:

```typescript
// BEFORE (WRONG):
import { InvalidSKUError } from '../errors/invalid-sku-error'

// AFTER (CORRECT):
import { InvalidSKUError } from '../errors'
```

**Pros**:
- Simple fix
- Maintains single errors file
- Standard barrel export pattern
- Less file overhead

**Cons**: None

### Option 2: Create Separate Error Files

Create individual error files instead of barrel:
- `errors/invalid-sku-error.ts`
- `errors/invalid-price-error.ts`
- `errors/invalid-inventory-error.ts`
- `errors/product-not-found-error.ts`

**Pros**:
- Matches import statements
- More granular

**Cons**:
- More files to maintain
- Unnecessary for simple error classes

---

## 🎯 Root Cause Analysis

This mismatch likely occurred because:

1. **Template inconsistency**: Value object templates use specific file imports
2. **Error generation pattern**: Error generation creates barrel export (`index.ts`)
3. **No validation**: Import paths not validated against actual file structure

The issue is in either:
- `/01-plan-layer-features` (planning stage generates wrong imports), OR
- `/03-generate-layer-code` (YAML generation creates wrong file structure)

---

## 📊 Impact Assessment

| Category | Impact | Severity |
|----------|--------|----------|
| **Compilation** | ❌ TypeScript errors | CRITICAL |
| **Runtime** | ❌ Module not found | CRITICAL |
| **RLHF Score** | -1 (RUNTIME ERROR) | HIGH |
| **Evaluation Score** | 2/5 (POOR) | HIGH |
| **Execution** | 🚫 Blocked (won't compile) | BLOCKER |

**This is a P0 BLOCKER** - prevents YAML execution entirely.

---

## ✅ Acceptance Criteria for Fix

1. ✅ Value object imports match actual error file structure
2. ✅ TypeScript compilation succeeds without "Module not found" errors
3. ✅ `/05-evaluate-layer-results` passes with Score 5 (PERFECT)
4. ✅ Expected RLHF score changes from -1 to +2
5. ✅ `/06-execute-layer-steps` can execute YAML without errors

---

## 🔗 Related

- **Dogfooding Experiment**: 003-modular-yaml-recovery.md
- **Phase**: Phase 5 (Evaluation)
- **Command**: `/05-evaluate-layer-results`
- **Related Issues**: #117 (modular YAML), #122 (execution dependencies)

---

## 📋 Next Steps

1. Identify where import paths are generated (`/01` templates vs `/03` YAML generation)
2. Apply Option 1 fix (change imports to use barrel exports)
3. Re-run `/05-evaluate-layer-results` to verify Score 5
4. Update dogfooding document with finding
5. Add regression test to prevent future mismatches

---

**Priority**: P0 (BLOCKER)
**Component**: Code Generation / Templates
**Discovered By**: Dogfooding Experiment 003 - `/05-evaluate-layer-results` command

🤖 Generated with [Claude Code](https://claude.com/claude-code)

Co-Authored-By: Claude <noreply@anthropic.com>

--- a/src/templates/backend-domain-template.regent
+++ b/src/templates/backend-domain-template.regent
@@ -1314,7 +1314,7 @@ metadata:
           __ERROR_NAME_PASCAL_CASE__Error,
           __USE_CASE_NAME_PASCAL_CASE__ValidationError,
           __USE_CASE_NAME_PASCAL_CASE__BusinessRuleError
-        } from '../errors/__USE_CASE_NAME_KEBAB_CASE__-errors'
+        } from '../errors'
 
         // Mock implementation for testing the interface contract
         class Mock__USE_CASE_NAME_PASCAL_CASE__ implements __USE_CASE_NAME_PASCAL_CASE__ {

--- a/src/templates/fullstack-domain-template.regent
+++ b/src/templates/fullstack-domain-template.regent
@@ -1950,7 +1950,7 @@ integration:
           __USE_CASE_NAME_PASCAL_CASE__ValidationError,
           __USE_CASE_NAME_PASCAL_CASE__NotFoundError,
           is__USE_CASE_NAME_PASCAL_CASE__Error
-        } from '../errors/__USE_CASE_NAME_KEBAB_CASE__-errors'
+        } from '../errors'
 
         /**
          * Shared domain layer tests for __USE_CASE_NAME_PASCAL_CASE__

--- a/src/templates/fullstack-infra-template.regent
+++ b/src/templates/fullstack-infra-template.regent
@@ -1563,7 +1563,7 @@ integration:
         import { PrismaClient, Prisma } from '@prisma/client'
         import type { __FEATURE_NAME_PASCAL_CASE__Repository } from '../../domain/repositories/__FEATURE_NAME_KEBAB_CASE__-repository'
         import type { __FEATURE_NAME_PASCAL_CASE__Entity } from '../../../__USE_CASE_NAME_KEBAB_CASE__/domain/entities/__USE_CASE_NAME_KEBAB_CASE__-entity'
-        import { __FEATURE_NAME_PASCAL_CASE__NotFoundError } from '../../../__USE_CASE_NAME_KEBAB_CASE__/domain/errors/__USE_CASE_NAME_KEBAB_CASE__-errors'
+        import { __FEATURE_NAME_PASCAL_CASE__NotFoundError } from '../../../__USE_CASE_NAME_KEBAB_CASE__/domain/errors'
 
         /**
          * Prisma implementation of __FEATURE_NAME_PASCAL_CASE__Repository
@@ -2185,7 +2185,7 @@ integration:
         import { describe, it, expect, beforeEach, vi, afterEach } from 'vitest'
         import { PrismaClient } from '@prisma/client'
         import { Prisma__FEATURE_NAME_PASCAL_CASE__Repository } from './prisma-__FEATURE_NAME_KEBAB_CASE__-repository'
-        import { __FEATURE_NAME_PASCAL_CASE__NotFoundError } from '../../../__USE_CASE_NAME_KEBAB_CASE__/domain/errors/__USE_CASE_NAME_KEBAB_CASE__-errors'
+        import { __FEATURE_NAME_PASCAL_CASE__NotFoundError } from '../../../__USE_CASE_NAME_KEBAB_CASE__/domain/errors'
 
         // Mock Prisma Client
         vi.mock('@prisma/client', () => ({

--- a/src/templates/parts/backend/steps/01-domain.part.regent
+++ b/src/templates/parts/backend/steps/01-domain.part.regent
@@ -182,7 +182,7 @@
           __ERROR_NAME_PASCAL_CASE__Error,
           __USE_CASE_NAME_PASCAL_CASE__ValidationError,
           __USE_CASE_NAME_PASCAL_CASE__BusinessRuleError
-        } from '../errors/__USE_CASE_NAME_KEBAB_CASE__-errors'
+        } from '../errors'
 
         // Mock implementation for testing the interface contract
         class Mock__USE_CASE_NAME_PASCAL_CASE__ implements __USE_CASE_NAME_PASCAL_CASE__ {

--- a/src/templates/parts/fullstack/steps/01-domain.part.regent
+++ b/src/templates/parts/fullstack/steps/01-domain.part.regent
@@ -466,7 +466,7 @@
           __USE_CASE_NAME_PASCAL_CASE__ValidationError,
           __USE_CASE_NAME_PASCAL_CASE__NotFoundError,
           is__USE_CASE_NAME_PASCAL_CASE__Error
-        } from '../errors/__USE_CASE_NAME_KEBAB_CASE__-errors'
+        } from '../errors'
 
         /**
          * Shared domain layer tests for __USE_CASE_NAME_PASCAL_CASE__

--- a/src/templates/parts/fullstack/steps/03-infra.part.regent
+++ b/src/templates/parts/fullstack/steps/03-infra.part.regent
@@ -79,7 +79,7 @@
         import { PrismaClient, Prisma } from '@prisma/client'
         import type { __FEATURE_NAME_PASCAL_CASE__Repository } from '../../domain/repositories/__FEATURE_NAME_KEBAB_CASE__-repository'
         import type { __FEATURE_NAME_PASCAL_CASE__Entity } from '../../../__USE_CASE_NAME_KEBAB_CASE__/domain/entities/__USE_CASE_NAME_KEBAB_CASE__-entity'
-        import { __FEATURE_NAME_PASCAL_CASE__NotFoundError } from '../../../__USE_CASE_NAME_KEBAB_CASE__/domain/errors/__USE_CASE_NAME_KEBAB_CASE__-errors'
+        import { __FEATURE_NAME_PASCAL_CASE__NotFoundError } from '../../../__USE_CASE_NAME_KEBAB_CASE__/domain/errors'
 
         /**
          * Prisma implementation of __FEATURE_NAME_PASCAL_CASE__Repository
@@ -701,7 +701,7 @@
         import { describe, it, expect, beforeEach, vi, afterEach } from 'vitest'
         import { PrismaClient } from '@prisma/client'
         import { Prisma__FEATURE_NAME_PASCAL_CASE__Repository } from './prisma-__FEATURE_NAME_KEBAB_CASE__-repository'
-        import { __FEATURE_NAME_PASCAL_CASE__NotFoundError } from '../../../__USE_CASE_NAME_KEBAB_CASE__/domain/errors/__USE_CASE_NAME_KEBAB_CASE__-errors'
+        import { __FEATURE_NAME_PASCAL_CASE__NotFoundError } from '../../../__USE_CASE_NAME_KEBAB_CASE__/domain/errors'
 
         // Mock Prisma Client
         vi.mock('@prisma/client', () => ({

